src/time.cpp

```diff
@@ -21,7 +21,7 @@ namespace g3 {
       // This is needed since latest version (at time of writing) of gcc4.7 does not implement this library function yet.
       // return value is SIMPLIFIED to only return a std::string
 
-      std::string put_time(const struct tm *tmb, const char *c_time_format) {
+      std::string put_time(const struct tm *tmb, const long nsec, const char *c_time_format) {
```

Comment:
The nsec parameter is not used. The reason for overloading std::put_time was that it was not implemented across platforms at the time of writing. It might be the function can be removed in another pull request but either way the API must not be changed

Ref: http://en.cppreference.com/w/cpp/io/manip/put_time